cli/azd/internal/version.go

```diff
@@ -6,31 +6,47 @@ package internal
 import (
 	"regexp"
 	"strings"
+
+	"github.com/blang/semver/v4"
 )
 
-// Version is the version string printed out by the `Version` command.
-// It's updated using ldflags in CI.
+// cDevVersionString is the default version that is used when [Version] is not overridden at build time, i.e.
+// a developer building locally using `go install`.
+const cDevVersionString = "0.0.0-dev.0 (commit 0000000000000000000000000000000000000000)"
+
+// The version string, as printed by `azd version`.
+//
+// This MUST be of the form "<semver> (commit <full commit hash>)"
+//
+// The default value here is used for a version built directly by a developer when running
+// `go install`
+//
+// Official build set this value based on the version and commit we are building, using `-ldflags`
+//
 // Example:
 //
-//	-ldflags="-X 'github.com/azure/azure-dev/cli/azd/internal.Version=0.0.1-alpha.1 (commit
+//	-ldflags="-X 'github.com/azure/azure-dev/cli/azd/internal.Version=0.0.1-alpha.1 (commit 8a49ae5ae9ab13beeade35f91ad4b4611c2f5574)'"
 //
-// 8a49ae5ae9ab13beeade35f91ad4b4611c2f5574)'"
-var Version = "0.0.0-dev.0 (commit 0000000000000000000000000000000000000000)"
-
-const UnknownVersion string = "unknown"
-const UnknownCommit string = "unknown"
+// This value is exported and not const so it can be mutated by certain tests. Instead of access this member
+// directly, use [VersionInfo] which returns a structured version of this value.
+//
+// nolint: lll
+var Version = cDevVersionString
 
-type VersionSpec struct {
-	Azd AzdVersionSpec `json:"azd"`
+func init() {
+	// VersionInfo panics if the version string is malformed, run the code at package startup to
```

Comment:
❤️ this change as well. These types of initialization panics really help us to assert invariants and avoid unexpected behavior at runtime